generators/kubernetes/templates/keycloak/keycloak.yml.ejs

```diff
@@ -0,0 +1,85 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: keycloak
+  namespace: <%= kubernetesNamespace %>
+  labels:
+    app: keycloak
+spec:
+  ports:
+  - name: http
+    port: 9080
+    targetPort: 8080
+  - name: https
+    port: 9443
+    targetPort: 8443
+  clusterIP: None
+  publishNotReadyAddresses: true  
+  selector:
+    app: keycloak
+---
+apiVersion: apps/v1
+kind: StatefulSet
```

Comment:
Shouldn't it be a Deployment? StatefulSet looks to be only viable for the DB

Reply:
@indiepopart Can you change this before I QA this PR?

Reply:
Would you please look at this [discussion](https://github.com/keycloak/keycloak/discussions/11763)?
`StatefulSet` is the recommended. I just tested `Deployment` and it works. Please confirm if it should be changed to `Deployment` anyways.

Reply:
ok lets keep statefulset then